src/file.js

```diff
@@ -153,6 +153,25 @@
   };
 
   /**
+   * Method to replace an extension, if one does not exist in the file string, it will be added.
+   *
+   * @param {string} fileString - File string to add the extension to if one does not exist
+   * @param {string} extension - Extension to add if one does not exist in `fileString`. The
+   *   value is the extension without the `.`. E.g. `js`, `html`.  Not `.js`, `.html`.
+   * @returns {string} New fileString with the new extension if one did not exist
+   */
+  File.replaceExtension = function(fileString, extension) {
+    var fileName  = File.parseFileName(fileString),
```

Comment:
This function can probably be more easily done with a single regex.  Something like this probably:

``` javascript
return fileString.replace(/([^.]+)$/, function(match) {
    return extension;
});
```

I like this website for testing [regex](https://regex101.com/)